### PR TITLE
Port CI to Github Actions and fix issues thus revealed

### PR DESCRIPTION
**Pull request by oremanj**
* Update to mypy 0.910, fix some errors due to newer typeshed plus TypeVarDef vs TypeVarLikeDef in plugin
* Change outcome.asend() and trio_typing.CompatAsyncGenerator to be compatible with newer typing.AsyncGenerator where an async generator's asend(), athrow(), aclose() return arbitrary awaitables rather than necessarily coroutines

### CODE REVIEW

**Comment by tjstum** [comment]
@altendky you might also take a look at this, since it'll obviate #29

**Comment by oremanj** [comment]
ah my bad, I didn't notice #29 before I started working on this! sorry :-(

File: ci.sh [hunk 1]
```diff
@@ -44,25 +44,18 @@ fi
...
-    # distutils. On Travis this apparently imports imp and triggers
-    # a deprecation warning.
-    if [ "$RUNTIME_ONLY" = "1" ]; then
-        pytest -W error -W ignore:::distutils -ra -v --pyargs trio_typing -k test_runtime
-    else
-        pytest -W error -W ignore:::distutils -ra -v -p trio_typing._tests.datadriven --pyargs trio_typing
-    fi
+# The data-driven tests import mypy.build, which imports
+# distutils. On Travis this apparently imports imp and triggers
+# a deprecation warning.
```
Comment (tjstum) [inline ci.sh:55]: does this still apply? does it still need to reference "travis"?

**Review by tjstum — APPROVED** [review]
lgtm! thanks

**Pull request merged**